src/gh_pr_upsert/core.py

```diff
@@ -1,2 +1,54 @@
-def hello_world():
-    return "Hello, world!"
+from gh_pr_upsert import git
+
+
+class PRUpsertError(Exception):
+    """Base class for all exceptions deliberately raised by this module."""
+
+    message = None
+    exit_status = None
+
+
+class OnDefaultBranchError(PRUpsertError):
+    message = "You must change to a branch before creating a PR"
+    exit_status = 2
```

Comment:
Different exit statuses are used for different reasons so that scripts that call `gh pr-upsert` can treat them differently.